### PR TITLE
Move node.*.key into node and rename process.key to name

### LINKED ISSUE #1534: Rename process.key to process.name
Rename process.key to process.name to better show that it's only used as an help for human to know what is the process same purpose as service.name.

### LINKED ISSUE #1535: Move Result.key, Event.key, Task.key, Map.key and Filter.key to Node.key
Move Result.key, Event.key, Task.key, Map.key and Filter.key to Node.key.

Every node type redeclare a parameter key. it should centralized in node.
this will require a bit of modification in the orchestrator (like passing the node.key as parameter of some functions) but nothing complicated.

### PR DESCRIPTION
**Pull request by krhubert**
close #1534 
close #1535

### CODE REVIEW

File: orchestrator/orchestrator.go [hunk 1]
```diff
@@ -121,12 +121,12 @@ func (s *Orchestrator) execute(filter func(wf *process.Process, node *process.Pr
 
 func (s *Orchestrator) executeNode(wf *process.Process, n *process.Process_Node, exec *execution.Execution, event *event.Event, data *types.Struct) error {
 	logrus.WithField("module", "orchestrator").
-		WithField("nodeID", n.ID()).
+		WithField("nodeID", n.Key).
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]:
```suggestion
		WithField("node.key", n.Key).
```

File: orchestrator/orchestrator.go [hunk 2]
```diff
@@ -215,7 +215,7 @@ func (s *Orchestrator) processTask(task *process.Process_Node_Task, wf *process.
 		ProcessHash:  wf.Hash,
 		EventHash:    eventHash,
 		ParentHash:   execHash,
-		StepID:       task.Key,
+		StepID:       nodeKey,
```
Comment (NicolasMahe) [inline orchestrator/orchestrator.go]: we should rename execution's StepID to stepKey.
@antho1404 what do you think?

File: process/graph.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ func (g Process) FindNodes(filter func(n *Process_Node) bool) []*Process_Node {
 // FindNode return a specific node in a graph identifies by its ID. Returns an error if there is no match or multiple matches
 func (g Process) FindNode(id string) (*Process_Node, error) {
```
Comment (NicolasMahe) [inline process/graph.go]:
```suggestion
func (g Process) FindNode(key string) (*Process_Node, error) {
```

File: orchestrator/orchestrator.go (hunk 2 above)
Comment (antho1404) [reply orchestrator/orchestrator.go]: I'm ok to change it, but why not `nodeKey` like that we always talk about the same thing.

Comment (NicolasMahe) [reply orchestrator/orchestrator.go]: `nodeKey` 👍

File: protobuf/types/process.proto [hunk 4]
```diff
@@ -138,37 +118,37 @@ message Process {
...
+    // Key that identifies the node.
+    string key = 1 [
+      (gogoproto.moretags) = 'hash:"name:1" validate:"required"'
+    ];
+
     oneof type {
       // Result is a trigger that listens for a specific result.
-      Result result = 1 [
+      Result result = 2 [
         (gogoproto.moretags) = 'hash:"name:1"'
```
Comment (antho1404) [inline protobuf/types/process.proto]: if you change the number in proto, don't forget to also change the hash
```suggestion
        (gogoproto.moretags) = 'hash:"name:2"'
```

File: protobuf/types/process.proto [hunk 5]
```diff
@@ -138,38 +118,38 @@ message Process {
...
+    // Key that identifies the node.
+    string key = 1 [
+      (gogoproto.moretags) = 'hash:"name:1" validate:"required"'
+    ];
+
     oneof type {
       // Result is a trigger that listens for a specific result.
-      Result result = 1 [
-        (gogoproto.moretags) = 'hash:"name:1"'
+      Result result = 2 [
```
Comment (NicolasMahe) [inline protobuf/types/process.proto]: i think you don't need to start from 0 in a oneof

**Comment by NicolasMahe** [comment]
Let's wait for issue https://github.com/mesg-foundation/engine/issues/1427 to be closed before merging this PR

File: protobuf/types/process.proto (hunk 5 above)
Comment (krhubert) [reply protobuf/types/process.proto]: from 0?

Comment (NicolasMahe) [reply protobuf/types/process.proto]: i mean from 1

**Pull request merged**